sympy/core/sympify.py

```diff
@@ -479,7 +479,7 @@ def sympify(a, locals=None, convert_xor=True, strict=False, rational=False,
     return expr
 
 
-def _sympify(a):
+def _sympify(a, expected_type = None):
```

Comment:
We could have `expected_type` default to `Basic` if it doesn't lead to a significant slowdown.

Reply:
Doing it basic would require change at every instance _sympify is used. None would give us a option that we want to check or not. Isn't None a better option.

Reply:
Any time the result is not Basic would be a bug.

Reply:
Okay